mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp

```diff
@@ -55,6 +58,52 @@ void createAttnTuningRangeBF(TuningParamSet *newSpace, AttentionOp attnOp,
   }
 }
 
+auto getOptinalSplitKFactors(RockGemmWrapperInterface gemmOp,
+                             int32_t gemmMPerBlock, int32_t gemmNPerBlock,
+                             int32_t gemmKPerBlock, int32_t kPack) {
+  auto info = PopulateParamsInfo::fromOp(gemmOp);
+  if (not info.gemmAType.isF32()) {
+    return std::vector<int64_t>{1};
+  }
+
+  struct LocalData {
+    int64_t splitKFactor{};
+    double workImbalance{};
+  };
+
+  const auto numCUs = gemmOp.getNumCU().value();
+  std::vector<LocalData> factors{};
+  for (int32_t splitKFactor = 1; splitKFactor < 100; ++splitKFactor) {
```

Comment:
Why 100? And is a unit step the right move here?

Reply:
I reduced to `32`. Regarding the unit step, I am doing filtering after that loop. I'm trying to select the variants which result in lowest work imbalance and then consider only 6 of them.

There is one more filter; the expected gain (based on the work imbalance regarding the default (data parallel) case) must be greater than 30%.

I would say these hard-coded constants are some hyper-parameters which require some consideration. We should discuss it.